versions/1/data_stream/manifest.spec.yml

```diff
@@ -83,6 +86,14 @@ spec:
             default: true
             examples:
               - false
+          url_allowed_schemes:
```

Comment:
I wonder if we should have a "generic" way to validate these kind of text fields. Maybe something like a regex pattern that Fleet policy editor can test the user input against (i.e. `regexPattern: /^https?/`). Although only URL scheme validation is introduced here, I can see other kinds of arbitrary validation being necessary. @elastic/fleet what do you think? It might be overengineered and could have security implications.

Reply:
Good point. I don't have a strong opinion, but I think that for common use cases as this one I would prefer to have explicit settings, using regexes is sometimes a bit error prone, for example `/^https?/` would check that the string starts with the correct scheme, but it could still be an invalid URL. We would still need to mark this as url with `type: url` to indicate Fleet to use an URL parser to validate it.

In any case, adding a `regexPattern` setting can make sense for other use cases too, so we could add it anyway.

This is also a bit related to [this comment](https://github.com/elastic/kibana/issues/116914#issuecomment-956208104). If we are going to have many validation options I would prefer something as the second option with "validators". If not I would prefer the simpler approach implemented by now here.

Reply:
I think we probably want to some "easy button" built-in validators for package developers so they don't have to think too much and we have consistent behavior across inputs. We can also ensure these are very well tested by Fleet so we don't have to put that burden on package developers.

The biggest benefit I can see to allowing packages specify a custom regex for validation would be that we can update it out-of-band from Stack releases. So if there's a bug in the regex, we can fix it quickly for users without requiring a Stack upgrade. That said, I don't think this is going to be a huge problem in practice and would prefer we keep things simple for now. As you mentioned, I believe there is some security risk as well with exposing a generic regex feature. We'd likely need some level of sanitization to avoid expensive, complex regexes that could be used to DDoS Kibana.

If we have other use cases for this in the future, we could reconsider.

Reply:
@joshdover so you are fine with the current approach with `type: url` and `url_allowed_scheme/protocol`?

Reply:
Just to add to this, I think the regex would be useful, I have noticed that for the custom logs integration we tell users not to use `-` in the dataset name but there is nothing to stop them from doing so at the moment. 
![Uploading Screenshot 2021-11-03 at 14.20.43.png…]()

Reply:
Good example. I am going to add this option too. I would still keep the option for allowed schemas because they cover a pretty common use case.

Reply:
I have started thinking a bit about the support for regular expressions, and there are some open questions. Apart of the mentioned risks, to make it useful in many cases we may need to negate regular expressions, what would require additional settings (for example in the mentioned example we would want to accept values that don't contain `-`).
Also, if we want to validate the regular expressions, we may find problems given that the package tooling is written in Go, but the regular expressions are going to be executed in Kibana.

So I am moving this to a separate PR for further discussion (https://github.com/elastic/package-spec/pull/245), and I think we can continue here with the original options.